geopandas/geodataframe.py

```diff
@@ -90,6 +90,18 @@ def __init__(self, *args, **kwargs):
                 geometry = "geometry"
 
         if geometry is not None:
+            if (
+                hasattr(geometry, "crs")
+                and geometry.crs
+                and crs
+                and geometry.crs != crs
+            ):
+                warnings.warn(
+                    "CRS mismatch between CRS of underlying GeometryArray passed as "
+                    "geometry and 'crs'. Use 'GeoDataFrame.crs = crs' to overwrite CRS "
```

Comment:
```suggestion
                    "CRS mismatch between CRS of the passed geometries "
                    "and 'crs'. Use 'GeoDataFrame.crs = crs' to overwrite CRS "
```

(the passed geometries can in principle also be a GeoSeries)